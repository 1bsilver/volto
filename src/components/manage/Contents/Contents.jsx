/**
 * Contents component.
 * @module components/manage/Contents/Contents
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Helmet from 'react-helmet';
import { Dropdown, Header, Menu, Icon, Input, Table } from 'semantic-ui-react';
import { Link } from 'react-router';

import { searchContent } from '../../../actions';
import { getBaseUrl } from '../../../helpers';

/**
 * ContentsComponent class.
 * @class ContentsComponent
 * @extends Component
 */
@connect(
  (state, props) => ({
    loaded: state.search.loaded,
    items: state.search.items,
    pathname: props.location.pathname,
  }),
  dispatch => bindActionCreators({ searchContent }, dispatch),
)
export default class ContentsComponent extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    searchContent: PropTypes.func.isRequired,
    loaded: PropTypes.bool.isRequired,
    items: PropTypes.arrayOf(
      PropTypes.shape({
        '@id': PropTypes.string,
        '@type': PropTypes.string,
        title: PropTypes.string,
        description: PropTypes.string,
      }),
    ),
    pathname: PropTypes.string.isRequired,
  };

  /**
   * Default properties.
   * @property {Object} defaultProps Default properties.
   * @static
   */
  static defaultProps = {
    items: [],
  };

  /**
   * Component will mount
   * @method componentWillMount
   * @returns {undefined}
   */
  componentWillMount() {
    this.props.searchContent(getBaseUrl(this.props.pathname), {
      'path.depth': 1,
      sort_on: 'getObjPositionInParent',
    });
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    if (!this.props.loaded) {
      return <span />;
    }

    return (
      <div id="page-contents">
        <Helmet title="Contents" />
        <div className="container">
          <article id="content">
            <h1>Contents</h1>
            <section id="content-core">
              <Menu stackable>
                <Menu.Menu>
                  <Menu.Item>
                    <Icon name="sort content ascending" /> Rearrange
                  </Menu.Item>
                  <Menu.Item><Icon name="upload" /> Upload</Menu.Item>

                </Menu.Menu>

                <Menu.Menu position="right">
                  <Menu.Item><Icon name="cut" /> Cut</Menu.Item>
                  <Menu.Item><Icon name="copy" /> Copy</Menu.Item>
                  <Menu.Item><Icon name="paste" /> Paste</Menu.Item>
                  <Menu.Item><Icon name="trash" /> Delete</Menu.Item>
                  <Dropdown
                    item
                    trigger={<span><Icon name="write" /> Batch</span>}
                  >
                    <Dropdown.Menu>
                      <Dropdown.Item>
                        <Icon name="text cursor" /> Rename
                      </Dropdown.Item>
                      <Dropdown.Item>
                        <Icon name="tags" /> Tags
                      </Dropdown.Item>
                      <Dropdown.Item>
                        <Icon name="random" /> State
                      </Dropdown.Item>
                      <Dropdown.Item>
                        <Icon name="edit" /> Properties
                      </Dropdown.Item>
                    </Dropdown.Menu>
                  </Dropdown>
                  <div className="ui right aligned category search item">
                    <div className="ui transparent icon input">
                      <input
                        className="prompt"
                        type="text"
                        placeholder="Filter..."
                      />
                      <i className="search link icon" />
                    </div>
                    <div className="results" />
                  </div>
                </Menu.Menu>
              </Menu>
              <Table selectable compact singleLine>
                <Table.Header>
                  <Table.Row>
                    <Table.HeaderCell>
                      <Dropdown
                        trigger={<Icon name="minus square" color="blue" />}
                        icon={null}
                      >
                        <Dropdown.Menu>
                          <Dropdown.Header content="Select..." />
                          <Dropdown.Item>
                            <Icon name="check square" color="blue" /> All
                          </Dropdown.Item>
                          <Dropdown.Item>
                            <Icon name="square outline" /> None
                          </Dropdown.Item>
                          <Dropdown.Divider />
                          <Dropdown.Header content="12 selected" />
                          <Input
                            icon="search"
                            iconPosition="left"
                            className="search"
                            placeholder="Filter..."
                          />
                          <Dropdown.Menu scrolling>
                            <Dropdown.Item>
                              <Icon name="delete" /> My first blog entry!
                            </Dropdown.Item>
                            <Dropdown.Item>
                              <Icon name="delete" /> My first blog entry!
                            </Dropdown.Item>
                            <Dropdown.Item>
                              <Icon name="delete" /> My first blog entry!
                            </Dropdown.Item>
                            <Dropdown.Item>
                              <Icon name="delete" /> My first blog entry!
                            </Dropdown.Item>
                            <Dropdown.Item>
                              <Icon name="delete" /> My first blog entry!
                            </Dropdown.Item>
                            <Dropdown.Item>
                              <Icon name="delete" /> My first blog entry!
                            </Dropdown.Item>
                            <Dropdown.Item>
                              <Icon name="delete" /> My first blog entry!
                            </Dropdown.Item>
                            <Dropdown.Item>
                              <Icon name="delete" /> My first blog entry!
                            </Dropdown.Item>
                            <Dropdown.Item>
                              <Icon name="delete" /> My first blog entry!
                            </Dropdown.Item>
                            <Dropdown.Item>
                              <Icon name="delete" /> My first blog entry!
                            </Dropdown.Item>
                            <Dropdown.Item>
                              <Icon name="delete" /> My first blog entry!
                            </Dropdown.Item>
                          </Dropdown.Menu>
                        </Dropdown.Menu>
                      </Dropdown>
                    </Table.HeaderCell>
                    <Table.HeaderCell width="6">Title</Table.HeaderCell>
                    <Table.HeaderCell width="5">Type</Table.HeaderCell>
                    <Table.HeaderCell width="5">Review state</Table.HeaderCell>
                    <Table.HeaderCell textAlign="right">
                      <Dropdown icon="ellipsis vertical">
                        <Dropdown.Menu className="left">
                          <Dropdown.Header content="Select columns to show" />
                          <Dropdown.Item>
                            <Icon name="square outline" /> ID
                          </Dropdown.Item>
                          <Dropdown.Item>
                            <Icon name="check square" color="blue" /> Title
                          </Dropdown.Item>
                          <Dropdown.Item>
                            <Icon name="check square" color="blue" /> Type
                          </Dropdown.Item>
                          <Dropdown.Item>
                            <Icon name="check square" color="blue" />
                            {' '}
                            Review state
                          </Dropdown.Item>
                          <Dropdown.Item>
                            <Icon name="square outline" /> Created on
                          </Dropdown.Item>
                          <Dropdown.Item>
                            <Icon name="square outline" /> Tags
                          </Dropdown.Item>
                          <Dropdown.Item>
                            <Icon name="square outline" /> Object Size
                          </Dropdown.Item>
                          <Dropdown.Item>
                            <Icon name="square outline" /> Creator
                          </Dropdown.Item>
                        </Dropdown.Menu>
                      </Dropdown>
                    </Table.HeaderCell>
                  </Table.Row>
                </Table.Header>
                <Table.Body>
                  {this.props.items.map(item => (
                    <Table.Row key={item['@id']}>
                      <Table.Cell verticalAlign="bottom">
                        <Icon name="check square" color="blue" />
                      </Table.Cell>
                      <Table.Cell>
                        <Link to="/"><Icon name="folder" /> {item.title}</Link>
                      </Table.Cell>
                      <Table.Cell>{item['@type']}</Table.Cell>
                      <Table.Cell>{item.review_state}</Table.Cell>
                      <Table.Cell textAlign="right">
                        <Dropdown icon="ellipsis vertical">
                          <Dropdown.Menu className="left">
                            <Dropdown.Item text="Edit" />
                            <Dropdown.Item text="View" />
                            <Dropdown.Divider />
                            <Dropdown.Item text="Cut" />
                            <Dropdown.Item text="Copy" />
                            <Dropdown.Item text="Delete" />
                            <Dropdown.Divider />
                            <Dropdown.Item text="Move to top of folder" />
                            <Dropdown.Item text="Move to bottom of folder" />
                          </Dropdown.Menu>
                        </Dropdown>
                      </Table.Cell>
                    </Table.Row>
                  ))}
                </Table.Body>
              </Table>
            </section>
          </article>
        </div>
      </div>
    );
  }
}
