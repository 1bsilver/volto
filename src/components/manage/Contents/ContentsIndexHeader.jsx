/**
 * Contents index header component.
 * @module components/manage/Contents/ContentsIndexHeader
 */

import React from 'react';
import PropTypes from 'prop-types';
import { DragSource, DropTarget } from 'react-dnd';

const widthValues = [
  'one',
  'two',
  'three',
  'four',
  'five',
  'six',
  'seven',
  'eight',
  'nine',
  'ten',
  'eleven',
  'twelve',
  'thirteen',
  'fourteen',
  'fiveteen',
  'sixteen',
];

/**
 * Component to display a folder contents index header.
 * @function ContentsIndexHeaderComponent
 * @returns {string} Markup of the component.
 */
export const ContentsIndexHeaderComponent = ({
  width,
  label,
  connectDragSource,
  connectDropTarget,
  isDragging,
}) =>
  connectDropTarget(
    connectDragSource(
      <th
        className={`${widthValues[width - 1]} wide`}
        style={{ opacity: isDragging ? 0.5 : 1, cursor: 'move' }}
      >
        {label}
      </th>,
    ),
  );

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
ContentsIndexHeaderComponent.propTypes = {
  /**
   * Width of the column
   */
  width: PropTypes.number.isRequired,
  /**
   * Label of the column
   */
  label: PropTypes.string.isRequired,
  /**
   * Handler to connect a drag source
   */
  connectDragSource: PropTypes.func.isRequired,
  /**
   * Handler to connect a drop source
   */
  connectDropTarget: PropTypes.func.isRequired,
  /**
   * Dragging state
   */
  isDragging: PropTypes.bool.isRequired,
  /**
   * Index number of the column
   */
  order: PropTypes.number.isRequired,
  /**
   * Action called when ordering a column
   */
  onOrderIndex: PropTypes.func.isRequired,
};

export default DropTarget(
  'index',
  {
    hover(props, monitor) {
      const dragOrder = monitor.getItem().order;
      const hoverOrder = props.order;

      if (dragOrder === hoverOrder) {
        return;
      }

      props.onOrderIndex(dragOrder, hoverOrder - dragOrder);

      monitor.getItem().order = hoverOrder;
    },
  },
  connect => ({
    connectDropTarget: connect.dropTarget(),
  }),
)(
  DragSource(
    'index',
    {
      beginDrag(props) {
        return {
          id: props.label,
          order: props.order,
        };
      },
    },
    (connect, monitor) => ({
      connectDragSource: connect.dragSource(),
      isDragging: monitor.isDragging(),
    }),
  )(ContentsIndexHeaderComponent),
);
