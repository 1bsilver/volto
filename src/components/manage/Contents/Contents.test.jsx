import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import Contents from './Contents';

const mockStore = configureStore();

jest.mock('../../theme/Pagination/Pagination', () =>
  jest.fn(() => <div className="Pagination" />),
);

jest.mock('moment', () =>
  jest.fn(() => ({
    format: jest.fn(() => 'Sunday, April 23, 2017 3:38 AM'),
    fromNow: jest.fn(() => 'a few seconds ago'),
  })),
);

describe('Contents', () => {
  it('renders a contents component', () => {
    const store = mockStore({
      search: {
        items: [
          {
            '@id': '/blog',
            '@type': 'Folder',
            title: 'Blog',
            descripton: 'My Blog',
            ModificationDate: '2017-04-19T22:48:56+02:00',
            EffectiveDate: '2017-04-19T22:48:56+02:00',
            review_state: 'private',
          },
        ],
        total: 1,
      },
      clipboard: {
        action: 'copy',
        source: ['/blog'],
        request: {
          loading: false,
          loaded: false,
        },
      },
      content: {
        delete: {
          loading: false,
          loaded: false,
        },
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <Contents location={{ pathname: '/blog' }} />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
