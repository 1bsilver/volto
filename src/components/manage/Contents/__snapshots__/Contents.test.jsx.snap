// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contents renders a contents component 1`] = `
<div
  id="page-contents"
>
  <div
    className="container"
  >
    <article
      id="content"
    >
      <div
        className="UploadModal"
      />
      <h1>
        Contents
      </h1>
      <section
        id="content-core"
      >
        <div
          className="ui stackable menu"
        >
          <div
            className="menu"
          >
            <a
              className="item"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="upload icon"
              />
               Upload
            </a>
          </div>
          <div
            className="right menu"
          >
            <a
              className="disabled item"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="cut icon"
              />
               Cut
            </a>
            <a
              className="disabled item"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="copy icon"
              />
               Copy
            </a>
            <a
              className="item"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="paste icon"
              />
               Paste
            </a>
            <a
              className="disabled item"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="trash icon"
              />
               Delete
            </a>
            <div
              aria-busy={undefined}
              aria-disabled={true}
              aria-expanded={false}
              aria-multiselectable={undefined}
              className="ui disabled item dropdown"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="listbox"
              tabIndex={-1}
            >
              <span>
                <i
                  aria-hidden="true"
                  className="write icon"
                />
                 Batch
              </span>
              <i
                aria-hidden="true"
                className="dropdown icon"
              />
              <div
                className="menu transition"
              >
                <div
                  aria-checked={undefined}
                  aria-disabled={undefined}
                  aria-selected={undefined}
                  className="item"
                  onClick={[Function]}
                  role="option"
                >
                  <i
                    aria-hidden="true"
                    className="text cursor icon"
                  />
                   Rename
                </div>
                <div
                  aria-checked={undefined}
                  aria-disabled={undefined}
                  aria-selected={undefined}
                  className="item"
                  onClick={[Function]}
                  role="option"
                >
                  <i
                    aria-hidden="true"
                    className="tags icon"
                  />
                   Tags
                </div>
                <div
                  aria-checked={undefined}
                  aria-disabled={undefined}
                  aria-selected={undefined}
                  className="item"
                  onClick={[Function]}
                  role="option"
                >
                  <i
                    aria-hidden="true"
                    className="random icon"
                  />
                   State
                </div>
                <div
                  aria-checked={undefined}
                  aria-disabled={undefined}
                  aria-selected={undefined}
                  className="item"
                  onClick={[Function]}
                  role="option"
                >
                  <i
                    aria-hidden="true"
                    className="edit icon"
                  />
                   Properties
                </div>
              </div>
            </div>
            <div
              className="ui right aligned category search item"
            >
              <div
                className="ui transparent icon input"
              >
                <div
                  className="ui input prompt"
                >
                  <input
                    onChange={[Function]}
                    placeholder="Filter..."
                    type="text"
                    value=""
                  />
                </div>
                <i
                  className="search link icon"
                />
              </div>
              <div
                className="results"
              />
            </div>
          </div>
        </div>
        <table
          className="ui selectable single line compact table"
        >
          <thead
            className=""
          >
            <tr
              className=""
            >
              <th
                className=""
              >
                <div
                  aria-busy={undefined}
                  aria-disabled={undefined}
                  aria-expanded={false}
                  aria-multiselectable={undefined}
                  className="ui simple dropdown"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  role="listbox"
                  tabIndex={0}
                >
                  <i
                    aria-hidden="true"
                    className="sort content ascending icon"
                  />
                  <div
                    className="menu transition"
                  >
                    <div
                      className="header"
                    >
                      Rearrange items by...
                    </div>
                    <div
                      aria-checked={undefined}
                      aria-disabled={undefined}
                      aria-selected={undefined}
                      className="item"
                      onClick={[Function]}
                      role="option"
                    >
                      <i
                        aria-hidden="true"
                        className="dropdown icon"
                      />
                      Id
                      <div
                        className="menu transition"
                      >
                        <div
                          aria-checked={undefined}
                          aria-disabled={undefined}
                          aria-selected={undefined}
                          className="item"
                          onClick={[Function]}
                          role="option"
                        >
                          <i
                            aria-hidden="true"
                            className="sort alphabet ascending icon"
                          />
                           
                          Ascending
                        </div>
                        <div
                          aria-checked={undefined}
                          aria-disabled={undefined}
                          aria-selected={undefined}
                          className="item"
                          onClick={[Function]}
                          role="option"
                        >
                          <i
                            aria-hidden="true"
                            className="sort alphabet descending icon"
                          />
                           
                          Descending
                        </div>
                      </div>
                    </div>
                    <div
                      aria-checked={undefined}
                      aria-disabled={undefined}
                      aria-selected={undefined}
                      className="item"
                      onClick={[Function]}
                      role="option"
                    >
                      <i
                        aria-hidden="true"
                        className="dropdown icon"
                      />
                      Title
                      <div
                        className="menu transition"
                      >
                        <div
                          aria-checked={undefined}
                          aria-disabled={undefined}
                          aria-selected={undefined}
                          className="item"
                          onClick={[Function]}
                          role="option"
                        >
                          <i
                            aria-hidden="true"
                            className="sort alphabet ascending icon"
                          />
                           
                          Ascending
                        </div>
                        <div
                          aria-checked={undefined}
                          aria-disabled={undefined}
                          aria-selected={undefined}
                          className="item"
                          onClick={[Function]}
                          role="option"
                        >
                          <i
                            aria-hidden="true"
                            className="sort alphabet descending icon"
                          />
                           
                          Descending
                        </div>
                      </div>
                    </div>
                    <div
                      aria-checked={undefined}
                      aria-disabled={undefined}
                      aria-selected={undefined}
                      className="item"
                      onClick={[Function]}
                      role="option"
                    >
                      <i
                        aria-hidden="true"
                        className="dropdown icon"
                      />
                      Publication date
                      <div
                        className="menu transition"
                      >
                        <div
                          aria-checked={undefined}
                          aria-disabled={undefined}
                          aria-selected={undefined}
                          className="item"
                          onClick={[Function]}
                          role="option"
                        >
                          <i
                            aria-hidden="true"
                            className="sort alphabet ascending icon"
                          />
                           
                          Ascending
                        </div>
                        <div
                          aria-checked={undefined}
                          aria-disabled={undefined}
                          aria-selected={undefined}
                          className="item"
                          onClick={[Function]}
                          role="option"
                        >
                          <i
                            aria-hidden="true"
                            className="sort alphabet descending icon"
                          />
                           
                          Descending
                        </div>
                      </div>
                    </div>
                    <div
                      aria-checked={undefined}
                      aria-disabled={undefined}
                      aria-selected={undefined}
                      className="item"
                      onClick={[Function]}
                      role="option"
                    >
                      <i
                        aria-hidden="true"
                        className="dropdown icon"
                      />
                      Created on
                      <div
                        className="menu transition"
                      >
                        <div
                          aria-checked={undefined}
                          aria-disabled={undefined}
                          aria-selected={undefined}
                          className="item"
                          onClick={[Function]}
                          role="option"
                        >
                          <i
                            aria-hidden="true"
                            className="sort alphabet ascending icon"
                          />
                           
                          Ascending
                        </div>
                        <div
                          aria-checked={undefined}
                          aria-disabled={undefined}
                          aria-selected={undefined}
                          className="item"
                          onClick={[Function]}
                          role="option"
                        >
                          <i
                            aria-hidden="true"
                            className="sort alphabet descending icon"
                          />
                           
                          Descending
                        </div>
                      </div>
                    </div>
                    <div
                      aria-checked={undefined}
                      aria-disabled={undefined}
                      aria-selected={undefined}
                      className="item"
                      onClick={[Function]}
                      role="option"
                    >
                      <i
                        aria-hidden="true"
                        className="dropdown icon"
                      />
                      Last modified
                      <div
                        className="menu transition"
                      >
                        <div
                          aria-checked={undefined}
                          aria-disabled={undefined}
                          aria-selected={undefined}
                          className="item"
                          onClick={[Function]}
                          role="option"
                        >
                          <i
                            aria-hidden="true"
                            className="sort alphabet ascending icon"
                          />
                           
                          Ascending
                        </div>
                        <div
                          aria-checked={undefined}
                          aria-disabled={undefined}
                          aria-selected={undefined}
                          className="item"
                          onClick={[Function]}
                          role="option"
                        >
                          <i
                            aria-hidden="true"
                            className="sort alphabet descending icon"
                          />
                           
                          Descending
                        </div>
                      </div>
                    </div>
                    <div
                      aria-checked={undefined}
                      aria-disabled={undefined}
                      aria-selected={undefined}
                      className="item"
                      onClick={[Function]}
                      role="option"
                    >
                      <i
                        aria-hidden="true"
                        className="dropdown icon"
                      />
                      Type
                      <div
                        className="menu transition"
                      >
                        <div
                          aria-checked={undefined}
                          aria-disabled={undefined}
                          aria-selected={undefined}
                          className="item"
                          onClick={[Function]}
                          role="option"
                        >
                          <i
                            aria-hidden="true"
                            className="sort alphabet ascending icon"
                          />
                           
                          Ascending
                        </div>
                        <div
                          aria-checked={undefined}
                          aria-disabled={undefined}
                          aria-selected={undefined}
                          className="item"
                          onClick={[Function]}
                          role="option"
                        >
                          <i
                            aria-hidden="true"
                            className="sort alphabet descending icon"
                          />
                           
                          Descending
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </th>
              <th
                className=""
              >
                <div
                  aria-busy={undefined}
                  aria-disabled={undefined}
                  aria-expanded={false}
                  aria-multiselectable={undefined}
                  className="ui dropdown"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  role="listbox"
                  tabIndex={0}
                >
                  <i
                    aria-hidden="true"
                    className="black square outline icon"
                  />
                  <div
                    className="menu transition"
                  >
                    <div
                      className="header"
                    >
                      Select...
                    </div>
                    <div
                      aria-checked={undefined}
                      aria-disabled={undefined}
                      aria-selected={undefined}
                      className="item"
                      onClick={[Function]}
                      role="option"
                    >
                      <i
                        aria-hidden="true"
                        className="blue check square icon"
                      />
                       All
                    </div>
                    <div
                      aria-checked={undefined}
                      aria-disabled={undefined}
                      aria-selected={undefined}
                      className="item"
                      onClick={[Function]}
                      role="option"
                    >
                      <i
                        aria-hidden="true"
                        className="square outline icon"
                      />
                       None
                    </div>
                    <div
                      className="divider"
                    />
                    <div
                      className="header"
                    >
                      0 selected
                    </div>
                    <div
                      className="ui left icon input search"
                    >
                      <i
                        aria-hidden="true"
                        className="search icon"
                      />
                      <input
                        placeholder="Filter..."
                        type="text"
                      />
                    </div>
                    <div
                      className="scrolling menu transition"
                    />
                  </div>
                </div>
              </th>
              <th
                className="four wide"
              >
                Title
              </th>
              <th
                className="four wide"
                style={
                  Object {
                    "cursor": "move",
                    "opacity": 1,
                  }
                }
              >
                Last modified
              </th>
              <th
                className="four wide"
                style={
                  Object {
                    "cursor": "move",
                    "opacity": 1,
                  }
                }
              >
                Publication date
              </th>
              <th
                className="four wide"
                style={
                  Object {
                    "cursor": "move",
                    "opacity": 1,
                  }
                }
              >
                Review state
              </th>
              <th
                className="right aligned"
              >
                <div
                  aria-busy={undefined}
                  aria-disabled={undefined}
                  aria-expanded={false}
                  aria-multiselectable={undefined}
                  className="ui dropdown"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  role="listbox"
                  tabIndex={0}
                >
                  <div
                    className="text"
                  />
                  <i
                    aria-hidden="true"
                    className="ellipsis vertical icon"
                  />
                  <div
                    className="menu transition left"
                  >
                    <div
                      className="header"
                    >
                      Select columns to show
                    </div>
                    <div
                      className="scrolling menu transition"
                    >
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="blue check square icon"
                        />
                        Last modified
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="blue check square icon"
                        />
                        Publication date
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="blue check square icon"
                        />
                        Review state
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        ID
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        Expiration date
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        Created on
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        Tags
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        Type
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        Folder
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        Excluded from navigation
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        Object Size
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        Last comment date
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        Total comments
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        End Date
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        Description
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        Creator
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        Location
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        UID
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        Start Date
                      </div>
                      <div
                        aria-checked={undefined}
                        aria-disabled={undefined}
                        aria-selected={undefined}
                        className="item"
                        onClick={[Function]}
                        role="option"
                      >
                        <i
                          aria-hidden="true"
                          className="square outline icon"
                        />
                        Type
                      </div>
                    </div>
                  </div>
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            className=""
          >
            <tr
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <td
                className=""
              >
                <i
                  aria-hidden="true"
                  className="grey content icon drag handle"
                />
              </td>
              <td
                className=""
              >
                <i
                  aria-hidden="true"
                  className="black square outline icon"
                  onClick={[Function]}
                  value="/blog"
                />
              </td>
              <td
                className=""
              >
                <a
                  onClick={[Function]}
                  style={Object {}}
                >
                  <i
                    aria-hidden="true"
                    className="file outline icon"
                  />
                   
                  Blog
                </a>
              </td>
              <td
                className=""
              >
                <span
                  title="Sunday, April 23, 2017 3:38 AM"
                >
                  a few seconds ago
                </span>
              </td>
              <td
                className=""
              >
                <span
                  title="Sunday, April 23, 2017 3:38 AM"
                >
                  a few seconds ago
                </span>
              </td>
              <td
                className=""
              >
                private
              </td>
              <td
                className="right aligned"
              >
                <div
                  aria-busy={undefined}
                  aria-disabled={undefined}
                  aria-expanded={false}
                  aria-multiselectable={undefined}
                  className="ui dropdown"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  role="listbox"
                  tabIndex={0}
                >
                  <div
                    className="text"
                  />
                  <i
                    aria-hidden="true"
                    className="ellipsis vertical icon"
                  />
                  <div
                    className="menu transition left"
                  >
                    <a
                      className="item"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      <i
                        aria-hidden="true"
                        className="write icon"
                      />
                       Edit
                    </a>
                    <a
                      className="item"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      <i
                        aria-hidden="true"
                        className="eye icon"
                      />
                       View
                    </a>
                    <div
                      className="divider"
                    />
                    <div
                      aria-checked={undefined}
                      aria-disabled={undefined}
                      aria-selected={undefined}
                      className="item"
                      onClick={[Function]}
                      role="option"
                    >
                      <i
                        aria-hidden="true"
                        className="cut icon"
                      />
                       Cut
                    </div>
                    <div
                      aria-checked={undefined}
                      aria-disabled={undefined}
                      aria-selected={undefined}
                      className="item"
                      onClick={[Function]}
                      role="option"
                    >
                      <i
                        aria-hidden="true"
                        className="copy icon"
                      />
                       Copy
                    </div>
                    <div
                      aria-checked={undefined}
                      aria-disabled={undefined}
                      aria-selected={undefined}
                      className="item"
                      onClick={[Function]}
                      role="option"
                    >
                      <i
                        aria-hidden="true"
                        className="trash icon"
                      />
                       Delete
                    </div>
                    <div
                      className="divider"
                    />
                    <div
                      aria-checked={undefined}
                      aria-disabled={undefined}
                      aria-selected={undefined}
                      className="item"
                      onClick={[Function]}
                      role="option"
                    >
                      <i
                        aria-hidden="true"
                        className="long arrow up icon"
                      />
                       
                      Move to top of folder
                    </div>
                    <div
                      aria-checked={undefined}
                      aria-disabled={undefined}
                      aria-selected={undefined}
                      className="item"
                      onClick={[Function]}
                      role="option"
                    >
                      <i
                        aria-hidden="true"
                        className="long arrow down icon"
                      />
                       
                      Move to bottom of folder
                    </div>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
        <div
          className="Pagination"
        />
      </section>
    </article>
  </div>
</div>
`;
